driver-sync/src/test/functional/com/mongodb/client/unified/EventMatcher.java

```diff
@@ -68,6 +68,16 @@ public void assertCommandEventsEquality(final String client, final BsonArray exp
                 }
             }
 
+            if (expected.containsKey("hasServerConnectionId")) {
+                boolean hasServerConnectionId = expected.getBoolean("hasServerConnectionId").getValue();
+                Integer serverConnectionId = actual.getConnectionDescription().getConnectionId().getServerValue();
```

Comment:
Make sure we're checking that there is a server value, since that's the thing that could be null.